src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    if (this.length === 0) {
+      return string;
+    }
```

Comment:
```suggestion
```
This check is redundant. If this.length === 0 then the loop will not even start and in the end, the function will still return `string`.
'string`, by the way, is a bad name for the variable. Use something more meaningful in the future

Reply:
Ok